### PR TITLE
(SIMP-6611) Document r10k deployment scenarios

### PR DESCRIPTION
**Pull request by op-ct**
This patch documents the Puppet deployment scenarios described in the
[SIMP Confluence][0], and refines them into subgroups that more
accurately depict the SIMP-relevant environment & scaling scenarios
users will need to consider from 6.4.0 and onwards.

The associated diagrams and verbiage have been updated accordingly.

SIMP-6611 #close

[0]: https://simp-project.atlassian.net/wiki/spaces/SD/pages/760774755/Puppet+deployment+scenarios,

### CODE REVIEW

File: docs/user_guide/SIMP_Administration/Environments/Deployment_Scenarios.inc [hunk 1]
```diff
@@ -1,64 +1,143 @@
...
+.. _ug-sa-env-deployment-scenarios--local:
+
+Local deployment scenarios
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Local deployment scenarios will work out-of-the-box on systems where SIMP
+has been installed by ISO or RPM.  You don't need a control repository, a
+remote Git-hosting service, or even a functioning network
+
+relies on SIMP RPM-created Puppet module Git repository
```
Comment (jeannegreulich) [inline docs/user_guide/SIMP_Administration/Environments/Deployment_Scenarios.inc]: This looks like it is missing something.  Should it just be "It releis on SIMP RPM..."  or was there something else you were going to say.

Comment (op-ct) [reply docs/user_guide/SIMP_Administration/Environments/Deployment_Scenarios.inc]: :ballot_box_with_check: Good catch; I've re-written that section

**Comment by op-ct** [comment]
> The file Environments/SIMP_Environments.inc isn't actually in ths one byut while I was reading through the entire section, the explanation "Infrastructure Environment didn't make sense. Is it just the site infrastructure? Why would that be in that table?

Yes, that is a different section from what this PR covers―but since you asked:

An "Infrastructure environment" isn't _just_ the site infrastructure―it's a slice of the site infrastructure that is considered its own "environment" (as is usually called something like "the production environment", "the staging environment", "the lab environment", etc.,).  This is what Gary Larizza calls ["internal environments" or  "application tiers"](http://garylarizza.com/blog/2014/03/26/random-r10k-workflow-ideas/) and close to what the Puppet docs now call ["Permanent test environments"](https://puppet.com/docs/puppet/5.5/environments_about.html#permanent-test-environments).

Infrastructure environments are the table because it's a classic source of confusion for administrators who are new to Puppet (and SIMP) when discussing environments.  They are generally used to talking about their _infrastructure's_ environments by just calling them "environments"―and don't always differentiate that from all the other Puppet/SIMP-specific "environments" that section discusses.

Our documentation needed to a.) make it clear that a distinction exists between classic infrastructure environments and all the _other_ kinds of environment we discuss in that section, and  b.) have a term we can refer to with a minimum of ambiguity.  So, we noted the difference on the diagram above and defined it in the "meta" environments section below.

**Comment by op-ct** [comment]
> Did you use .inc (instead of .rst and a menu in Environemnts) to force them to read all the way through?

The subsections have always been .inc in the "Deploying SIMP Environments" section, which is consistent with how all other sections at that menu-depth include their content.

We can change the files to .rst if that would be better, but I'd rather leave that kind of plumbing to another PR.

**Pull request merged**